LEGO1/mxpalette.cpp

```diff
@@ -1,4 +1,351 @@
+  { 227u, 232u, 236u, 0u },
+  { 224u, 221u, 209u, 0u },
+  { 215u, 222u, 215u, 0u },
+  { 213u, 214u, 215u, 0u },
+  { 214u, 214u, 203u, 0u },
+  { 255u, 219u, 57u, 0u },
+  { 206u, 206u, 206u, 0u },
+  { 206u, 206u, 198u, 0u },
+  { 255u, 214u, 18u, 0u },
+  { 207u, 203u, 186u, 0u },
+  { 197u, 199u, 199u, 0u },
+  { 255u, 206u, 0u, 0u },
+  { 207u, 198u, 159u, 0u },
+  { 247u, 204u, 0u, 0u },
+  { 189u, 198u, 189u, 0u },
+  { 189u, 189u, 189u, 0u },
+  { 238u, 199u, 0u, 0u },
+  { 189u, 189u, 181u, 0u },
+  { 238u, 190u, 24u, 0u },
+  { 181u, 189u, 184u, 0u },
+  { 161u, 186u, 224u, 0u },
+  { 181u, 181u, 181u, 0u },
+  { 231u, 189u, 0u, 0u },
+  { 173u, 182u, 173u, 0u },
+  { 222u, 181u, 0u, 0u },
+  { 173u, 173u, 173u, 0u },
+  { 213u, 182u, 0u, 0u },
+  { 172u, 173u, 160u, 0u },
+  { 214u, 173u, 0u, 0u },
+  { 165u, 165u, 165u, 0u },
+  { 206u, 173u, 0u, 0u },
+  { 160u, 168u, 151u, 0u },
+  { 206u, 164u, 0u, 0u },
+  { 198u, 165u, 0u, 0u },
+  { 157u, 156u, 156u, 0u },
+  { 134u, 156u, 200u, 0u },
+  { 153u, 156u, 144u, 0u },
+  { 142u, 156u, 161u, 0u },
+  { 189u, 156u, 0u, 0u },
+  { 148u, 148u, 148u, 0u },
+  { 146u, 148u, 138u, 0u },
+  { 133u, 143u, 161u, 0u },
+  { 189u, 143u, 0u, 0u },
+  { 140u, 140u, 140u, 0u },
+  { 177u, 147u, 0u, 0u },
+  { 131u, 140u, 136u, 0u },
+  { 146u, 130u, 126u, 0u },
+  { 170u, 137u, 0u, 0u },
+  { 132u, 132u, 130u, 0u },
+  { 123u, 125u, 125u, 0u },
+  { 123u, 123u, 133u, 0u },
+  { 153u, 126u, 0u, 0u },
+  { 114u, 116u, 118u, 0u },
+  { 110u, 112u, 108u, 0u },
+  { 97u, 109u, 136u, 0u },
+  { 127u, 108u, 6u, 0u },
+  { 0u, 173u, 0u, 0u },
+  { 100u, 99u, 101u, 0u },
+  { 176u, 71u, 41u, 0u },
+  { 36u, 142u, 33u, 0u },
+  { 98u, 91u, 75u, 0u },
+  { 80u, 88u, 104u, 0u },
+  { 252u, 0u, 0u, 0u },
+  { 78u, 71u, 73u, 0u },
+  { 73u, 71u, 78u, 0u },
+  { 62u, 63u, 61u, 0u },
+  { 0u, 66u, 211u, 0u },
+  { 99u, 51u, 14u, 0u },
+  { 198u, 0u, 0u, 0u },
+  { 189u, 0u, 0u, 0u },
+  { 0u, 57u, 206u, 0u },
+  { 181u, 0u, 0u, 0u },
+  { 0u, 56u, 185u, 0u },
+  { 173u, 0u, 0u, 0u },
+  { 165u, 0u, 0u, 0u },
+  { 49u, 49u, 49u, 0u },
+  { 0u, 49u, 165u, 0u },
+  { 156u, 0u, 0u, 0u },
+  { 42u, 45u, 60u, 0u },
+  { 148u, 0u, 0u, 0u },
+  { 140u, 0u, 0u, 0u },
+  { 41u, 41u, 41u, 0u },
+  { 0u, 41u, 144u, 0u },
+  { 132u, 0u, 0u, 0u },
+  { 123u, 0u, 0u, 0u },
+  { 7u, 35u, 114u, 0u },
+  { 34u, 36u, 32u, 0u },
+  { 115u, 0u, 0u, 0u },
+  { 107u, 0u, 0u, 0u },
+  { 90u, 0u, 0u, 0u },
+  { 23u, 24u, 27u, 0u },
+  { 74u, 0u, 0u, 0u },
+  { 15u, 15u, 16u, 0u },
+  { 49u, 0u, 0u, 0u },
+  { 16u, 12u, 4u, 0u },
+  { 7u, 8u, 8u, 0u },
+  { 0u, 0u, 8u, 0u },
+  { 255u, 251u, 240u, 0u },
+  { 160u, 160u, 164u, 0u },
+  { 128u, 128u, 128u, 0u },
+  { 255u, 0u, 0u, 0u },
+  { 0u, 255u, 0u, 0u },
+  { 255u, 255u, 0u, 0u },
+  { 0u, 0u, 255u, 0u },
+  { 255u, 0u, 255u, 0u },
+  { 0u, 255u, 255u, 0u },
+  { 255u, 255u, 255u, 0u }
+};
+
+// GLOBAL: LEGO1 0x10102188 1024
+LPPALETTEENTRY g_defaultPalette;
```

Comment:
Hmm what is this variable? You've marked it with the same offset as the array above.

Reply:
/shrug I thought it was like a struct how after listing the structure you give it a name, though it's probably just initializing an empty static thing. You could probably remove this.

Reply:
> though it's probably just initializing an empty static thing

No, it's the identifier for the array itself. I suspect the confusion comes from the fact that `T[]` can be used as `T*` without conversion, and not verifying the data itself at the assembler level.